component/site/libraries/helper.php

```diff
@@ -1384,6 +1386,20 @@ function isEventEditor()
 			else if ($user->canedit)
 			{
 				$isEventEditor = true;
+			} else if ($user->cancreate) {
```

Comment:
This is not ideal - you are in a sense passing a hidden parameter in here.  Better would be a separate canEditEventById method.  I'll let it through for now though